src/Resources/skeleton/forgottenPassword/ForgottenPasswordController.tpl.php

```diff
@@ -0,0 +1,110 @@
+{
+    /**
+     * @Route("/forgotten-password", name="app_forgotten_password")
+     */
+    public function request(Request $request, \Swift_Mailer $mailer): Response
+    {
+        $form = $this->createForm(<?= $request_form_class_name ?>::class);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $user = $this->getDoctrine()->getRepository(<?= $user_class_name ?>::class)->findOneBy([
+                '<?= $email_field ?>' => $form->get('<?= $email_field ?>')->getData(),
+            ]);
+
+            // do not reveal whether an user account was found or not
+            if (!$user) {
+                return $this->redirectToRoute('app_check_email');
+            }
+
+            // generate a reset password token, that the user could use to change his password
+            $resetPasswordToken = new <?= $token_class_name?>($user);
+            $this->getDoctrine()->getManager()->persist($resetPasswordToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            $message = (new \Swift_Message('Your password reset request'))
+                ->setFrom(['noreply@mydomain.com' => 'Noreply'])
+                ->setTo($user-><?= $email_getter ?>())
+                ->setBody($this->renderView('forgottenPassword/email.txt.twig', [
+                    'token' => $resetPasswordToken,
+                ]))
+            ;
+            $mailer->send($message);
+
+            return $this->redirectToRoute('app_check_email');
+        }
+
+        return $this->render('forgottenPassword/request.html.twig', [
+            'requestForm' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/check-email", name="app_check_email")
+     */
+    public function checkEmail()
+    {
+        return $this->render('forgottenPassword/check_email.html.twig', [
+            'tokenLifetime' => <?= $token_class_name ?>::TTL_HOURS,
+        ]);
+    }
+
+    /**
+     * @Route("/reset-password/{tokenAndSelector}", name="app_reset_password")
+     */
+    public function reset(Request $request, $tokenAndSelector, UserPasswordEncoderInterface $passwordEncoder): Response
+    {
+        $passwordResetToken = $this->getDoctrine()->getRepository(PasswordResetToken::class)->findOneBy([
+            'selector' => substr($tokenAndSelector, 0, PasswordResetToken::SELECTOR_LENGTH),
+        ]);
+
+        if (!$passwordResetToken) {
+            throw $this->createNotFoundException();
+        }
+        if ($passwordResetToken->isExpired() || !$passwordResetToken->isTokenEquals(substr($tokenAndSelector, PasswordResetToken::SELECTOR_LENGTH))) {
+            $this->getDoctrine()->getManager()->remove($passwordResetToken);
+            $this->getDoctrine()->getManager()->flush();
+
+            throw $this->createNotFoundException();
+        }
+
+        $form = $this->createForm(<?= $resetting_form_class_name ?>::class);
```

Comment:
`$form = $this->createForm(<?= $resetting_form_class_name ?>::class, $passwordResetToken->getUser());`

Important to bind data on the user, otherwise the form will create a new user :p

Reply:
Actually, that's complex case, because we bind form to `User::class`, and on `isValid()` call it says that user object is invalid if you have validation constraints on `User`. I think we should either keep for binding to `User` and pass the user object to the form as the 2nd arg, or do not bind the form on User object at all

Reply:
Really? The form has no `data_class`, I was expecting it to be bound to an array.

Reply:
I deleted my previous comment, I found where the binding was done. 
I prefer not to bind the form to anything, but to an array, as I don't see any scenario where we may need to access/validate `user` from the form, we just want to be able to do `getData()`

Reply:
Modification done